src/prefect/core/flow.py

```diff
@@ -1577,6 +1578,10 @@ def register(
                 Flow's project and name will be used.
             - no_url (bool, optional): if `True`, the stdout from this function will not
                 contain the URL link to the newly-registered flow in the Cloud UI
+            - idempotency_key (str, optional): an optional idempotency key for this flow
```

Comment:
I like the docstring you put on the client; could you duplicate that here?

Reply:
Yeah I can do that. I tried to make each specific to it's scope but I agree that one is better :)